### ISSUE TITLE
Carrot monitor breaks when the queue from a completed message log gets removed from the config

### ISSUE THREAD

**Issue by chris104957** [chris104957/django-carrot#81]
traceback
```
Internal Server Error: /carrot/api/message-logs/completed/
django_1  | Traceback (most recent call last):
django_1  |   File "/usr/local/lib/python3.5/dist-packages/carrot/utilities.py", line 50, in get_host_from_name
django_1  |     queue_host = list(filter(lambda queue: queue['name'] == name, queues))[0]['host']
django_1  | IndexError: list index out of range
django_1  | 
django_1  | During handling of the above exception, another exception occurred:
django_1  | 
django_1  | Traceback (most recent call last):
django_1  |   File "/usr/local/lib/python3.5/dist-packages/django/core/handlers/exception.py", line 35, in inner
django_1  |     response = get_response(request)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/django/core/handlers/base.py", line 128, in _get_response
django_1  |     response = self.process_exception_by_middleware(e, request)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/django/core/handlers/base.py", line 126, in _get_response
django_1  |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
django_1  |     return view_func(*args, **kwargs)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/viewsets.py", line 95, in view
django_1  |     return self.dispatch(request, *args, **kwargs)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/views.py", line 494, in dispatch
django_1  |     response = self.handle_exception(exc)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/views.py", line 454, in handle_exception
django_1  |     self.raise_uncaught_exception(exc)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/views.py", line 491, in dispatch
django_1  |     response = handler(request, *args, **kwargs)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/mixins.py", line 45, in list
django_1  |     return self.get_paginated_response(serializer.data)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/serializers.py", line 742, in data
django_1  |     ret = super(ListSerializer, self).data
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/serializers.py", line 262, in data
django_1  |     self._data = self.to_representation(self.instance)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/serializers.py", line 660, in to_representation
django_1  |     self.child.to_representation(item) for item in iterable
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/serializers.py", line 660, in <listcomp>
django_1  |     self.child.to_representation(item) for item in iterable
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/serializers.py", line 491, in to_representation
django_1  |     attribute = field.get_attribute(instance)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/fields.py", line 441, in get_attribute
django_1  |     return get_attribute(instance, self.source_attrs)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/rest_framework/fields.py", line 100, in get_attribute
django_1  |     instance = getattr(instance, attr)
django_1  |   File "/usr/local/lib/python3.5/dist-packages/carrot/models.py", line 66, in virtual_host
django_1  |     return str(get_host_from_name(self.queue))
django_1  |   File "/usr/local/lib/python3.5/dist-packages/carrot/utilities.py", line 59, in get_host_from_name
django_1  |     raise Exception('Cannot find queue called %s in settings.CARROT queue list' % name)
django_1  | Exception: Cannot find queue called default in settings.CARROT queue list

```

**Comment by chris104957** [comment]
To fix 
- [x] Create a carrot exceptions module, with a `CarrotConfigException`
- [x] Raise `CarrotConfigException` when the above error occurs
- [x] In the `MessageLog` model, in the `virtual_host` property method, catch the exception and return `None` if it occurs

**Comment by chris104957** [comment]
Fixed for 1.2
